### PR TITLE
fix: room creation flow and join on invites

### LINKED ISSUE #69: Users are not creating rooms and joining correctly
Only works sending messages when users already have friends

### PR DESCRIPTION
**Pull request by guidota**
## Description

- Fixes room aliases
- Add room to the list when created
- Improve Error handling
- Avoid picking the same user more than once in a tick
- Avoid users adding themselves as friend
- Better debug logging

Fixes #69

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -142,11 +142,11 @@ impl Client {
         )
         .await
         .expect("Couldn't create client");
-        self.inner = Arc::new(Mutex::new(client));
+        self.inner = Arc::new(client);
```
Comment (guidota) [inline src/client.rs:145]: No need to have a mutex here since `Client` has internal mutexes to handle everything "safe" (or at least I expect)

Comment (moliva) [reply src/client.rs:145]: Did you test this already?

Comment (guidota) [reply src/client.rs:145]: yes, locally and now the workflow is running based on this branch.

File: src/client.rs [hunk 2]
```diff
@@ -193,40 +193,60 @@ impl Client {
...
-            let joined_rooms = res.rooms.join.keys().cloned().collect::<Vec<_>>();
-            let invited_rooms = res.rooms.invite.keys().cloned().collect::<Vec<_>>();
-            SyncResult::Ok {
-                joined_rooms,
-                invited_rooms,
-                cancel_sync,
-            }
-        } else {
-            SyncResult::Failed
+        let res = response.unwrap();
```
Comment (moliva) [inline src/client.rs:243]: Let's change this to use `Result#expect` instead with a short description.

File: src/client.rs [hunk 3]
```diff
@@ -235,7 +255,7 @@ impl Client {
     /// If room_id is not one of the joined rooms or couldn't retrieve it.
     ///
     pub async fn send_message(&self, room_id: &RoomId, message: String) {
-        let client = self.inner.lock().await;
+        let client = &self.inner;
```
Comment (moliva) [inline src/client.rs:258]: Is the reference `&` still needed if `self` is already used as a shared reference here (also applies to previous changes as well)?

File: src/client.rs [hunk 4]
```diff
@@ -438,9 +468,6 @@ async fn on_room_message(
             if event.sender.localpart() == user_id.localpart() {
                 return;
             }
-            room.read_receipt(&event.event_id)
```
Comment (moliva) [inline src/client.rs:441]: Why removing this?

File: src/client.rs (hunk 3 above)
Comment (guidota) [reply src/client.rs:258]: it is. the reason is that `self.inner` doesn't implement the `Copy` trait and the compiler complains about this move.

I think I can remove the `Arc` and use it as it is (without local vars)

File: src/client.rs (hunk 4 above)
Comment (guidota) [reply src/client.rs:441]: it was panicking, will revisit this later in another PR.

Comment (moliva) [reply src/client.rs:441]: Let's add a TODO here if that's the case

Comment (guidota) [reply src/client.rs:457]: I think it's more than just that line so I filed this issue #70.

**Review by moliva — APPROVED** [review]

**Pull request merged**